dojo.py

```diff
@@ -218,3 +219,78 @@ def print_unallocated(self, filename=None):
                 file.write("\n" + "------------------------------------------\n")
                 file.write("{}".format(str1) + "\n")
                 file.close()
+
+    def reallocate_person(self, first_name, last_name, new_room_name):
+        person_identifier = first_name + " " + last_name
+
+        if person_identifier not in [person.person_name for person in self.total_people]:
+            return "Sorry that person does not exist"
+
+        if new_room_name not in [room.room_name for room in self.total_rooms]:
+            return "Sorry that room does not exist"
+
+        for person in self.total_people:
+            if person_identifier == person.person_name:
+                person_reallocating = person
+
+        rooms_occupied = []
+
+        for room in self.total_rooms:
+            if room.room_name == new_room_name:
+                room_to_relocate = room
+            if person_reallocating in room.occupants:
+                rooms_occupied.append(room)
+
+        if len(room_to_relocate.occupants) == room_to_relocate.maximum_occupants:
+            return "sorry that room is full"
+
+        else:
+            if room_to_relocate.type == "office":
+                if person_reallocating in self.waiting_for_office_allocation:
+                    self.waiting_for_office_allocation.remove(person_reallocating)
+                    room_to_relocate.occupants.append(person_reallocating)
+                    return "You has been removed from office waiting list to {0} {1}" \
```

Comment:
This sentence is a bit odd in my opinion. If it started with the person in the office list this would help.